src/generic/datavgen.cpp

```diff
@@ -1724,6 +1730,7 @@ wxDataViewMainWindow::wxDataViewMainWindow( wxDataViewCtrl *parent, wxWindowID i
     m_useCellFocus = false;
     m_currentRow = (unsigned)-1;
     m_lineHeight = GetDefaultRowHeight();
+    m_heightCache = new wxHashTable();
```

Comment:
Oh i mean the cache really helps currently. But you are right i am concerned too about the cache size when having a lot of items in the control. To use the cache only for the visible items might not work. It is especially necessary for all the items above the first visible item. 
Let's say i scrolled down the ScrolledWindow to y-position 100000. To repaint the control we need to know which item is the first visible item. See line 2007-2008:
```
    // compute which items needs to be redrawn
    unsigned int item_start = GetLineAt( wxMax(0,update.y) );
```
GetLineAt starts at y-position 0 and item 0, queries all cell renderers of each row for its height. If the loop reaches the target y-position (for instance 100000) it finally knows the first item to draw.. Without cache this loop takes almost 5 (five) seconds which is needed for EVERY repaint. With cache it needs a few milliseconds.